Design.md

```diff
@@ -0,0 +1,241 @@
+  + [getInformation](#getinformation)
+  + [updateOsisCaps (?)](#updateosiscaps----)
+  + [getBucketList](#getbucketlist)
+  + [getOsisUsage](#getosisusage)
+* [S3 Operations Example Activity Diagram](#s3-operations-example-activity-diagram)
+</details>
+
+## OSIS inital setup
+
+For the initial installation of OSIS application, following steps must be executed.
+1. A **Super Admin** account will be created as part of S3C/Vault setup
+1. Set OSE properties:
+   1. S3 URL
+   1. Admin Vault URL along with credentials as Super Admin access key/ secret key
+    
+_**Note: "Super Admin" account has full Account Management access and can use "AssumeRoleBackbeat" to assume a role of another Account.**_
+
+## Tenant APIs
+<u>**Important Assumptions:**</u>
+* All the Tenant APIs on OSIS will use **Super Admin credentials** to manage on Vault.
+* A Vault Account is equivalent of OSE Tenant.
+
+### Create Tenant
+This API creates a Tenant on Vault.  
+1. Vault `create-account` api will be called using vaultclient. 
+    * Tenant Name is stored as Account Name  
+    * `cd_tenant_ids` list as part of the request to be stored as a property of the vault account.
+    * Account ID to be sent back as storage tenant ID to OSE
+1. `generate-account-access-key` for the account using vaultclient.
+1. Create a role with
+   1. Role name : `osis` (So role-arn can be generated in the format `arn:aws:iam::[account-id]:role/osis`
+   1. Trust policy
+   ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [{
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "service-name"
+            },
+            "Action": "sts:AssumeRole"
+            }]
+    }
+   ```
+1. `attach-role-policy` with s3 and iam full access to this role
+1. `delete-access-key` for the account.
+
+### Create Tenant Example Activity Diagram
+![Create Tenant](Design_Files/OSE-OSIS-Create-Tenant.png)
+
+### Query Tenants
+This API will query tenants on Vault using a `filter` parameter.
+1. Vault `list-accounts` api will be called using vaultclient.
+1. Filter the result accounts using the `filter` parameter in the request
+
+### List Tenants
+This API will list tenants on Vault. Vault `list-accounts` api will be called using vaultclient.
+
+### Get Tenant
+This API will return the tenant on Vault with `tenantID`.
+1. Vault `get-account` api will be called using vaultclient with the provided `tenantID` as `AccountID`.
+
+### Head Tenant
+This API will check whether the tenant exists on Vault with `tenantID`.
+1. Vault `get-account` api will be called using vaultclient with the provided `tenantID` as `AccountID`.
+1. Return `true` or `false` if result is returned or not respectively.
+
+### Delete Tenant
+This API will delete the tenant on Vault. 
+
+**Approach 1:**
+1. Vault `delete-account` api will be called using vaultclient with the provided `tenantID` as `AccountID`.
+1. Return error if account is not empty.
+
+**Approach 2:** [suggested]
```

Comment:
Suggested by whom?
just confirming. :)

I know this is still an open question. If we are documenting `[suggested]`, we need to mention the source as well.

Reply:
suggested by me :) 
I will removed [suggestd] part

Reply:
Updated cdTenantIdList in commit 59a5a88